src/openpgp.js

```diff
@@ -311,6 +312,21 @@ export function decrypt({ message, privateKeys, passwords, sessionKeys, publicKe
     result.data = format === 'binary' ? decrypted.getLiteralData() : decrypted.getText();
     result.filename = decrypted.getFilename();
     linkStreams(result, message);
+    if (expectSigned) {
+      if (publicKeys.length === 0) {
+        throw new Error('public keys are required to verify message signatures');
+      }
+      if (result.signatures.length === 0) {
+        throw new Error('message is not signed');
+      }
+      result.data = stream.concat([
+        result.data,
+        stream.fromAsync(() => util.anyPromise(
+          result.signatures.map(sig => sig.verified)
+        ).then(() => {}) // append nothing to the data
+          .catch(() => { throw new Error('invalid message signature'); }))
```

Comment:
I know this was my suggestion but probably it's cleaner to write it like
```suggestion
        stream.fromAsync(async () => {
          await util.anyPromise(result.signatures.map(sig => sig.verified));
        })
```